interfaces/repo.go

```diff
@@ -34,23 +34,23 @@ type Repository struct {
 	m      sync.Mutex
 	ifaces map[string]Interface
 	// Indexed by [snapName][plugName]
-	plugs map[string]map[string]*Plug
-	slots map[string]map[string]*Slot
+	plugs map[string]map[string]*snap.PlugInfo
+	slots map[string]map[string]*snap.SlotInfo
 	// given a slot and a plug, are they connected?
-	slotPlugs map[*Slot]map[*Plug]*Connection
+	slotPlugs map[*snap.SlotInfo]map[*snap.PlugInfo]*Connection
```

Comment:
not a blocker here, but I am a bit vary of using *Plug/SlotInfo here, maybe the data structure will change completely that it will not be relevant, but we are using *Plug/SlotInfo as they were unique but they aren't globally, only inside the repo because we are carefully to get them out of Repo.plugs and Repo.slots but that's fragile

Reply:
We could change this to

`slotPlugs map[string]map[string]*Connection` where the first string is `slotRef.String()` and the 2nd string is `plugRef.String()`

Reply:
I'd keep this as is at this point. In the upcoming branches I'm introducing ConnectedPlug/ConnectedSlot and can revisit this aspect.